### PR TITLE
USWDS: Banner - Remove grid-layout dependency. Resolves #4868

### LINKED ISSUE #4868: Banner remove grid layout dependency
Remove layout dependency from USA Banner. The banner package should only compile necessary banner styles to preserve look and feel. It **should not** compile any CSS layout class utilities.

### PR DESCRIPTION
**Pull request by mahoneycm**
## Component 
[Preview →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-usa-banner-package/iframe.html?args=&id=components-banner--default&viewMode=story)

## Description

This PR removes the grid-layout dependency to cut down on the component's package size. It uses flex to maintain the same look as before.

Resolves #4868 

### Isolate Banner package
- In `uswds/_index.scss` remove all packages besides `usa-banner`
- Remove `/src/styles` so it only reads `@forward "usa-banner"`

### Check output css file size
- Run `npx gulp buildSass` to compile CSS
- Check `dist/css/uswds.css` filesize _(I use a vscode plugin but you can cd the directory to it and run `open .`_
- With this change it should be around 17kb. Before it was about 47kb.

If you do the above steps on another branch, you should see the larger filesize.

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Review by mejiaj — CHANGES_REQUESTED** [review]
@mahoneycm thanks for taking this issue on. I can see layout grid dependency was removed and tested the output. I've noticed on larger screens (approximately 1200px and higher) there's a visual discrepancy to the current banner.

[cm-usa-banner-package (top)](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-usa-banner-package/iframe.html?args=&id=components-banner--default&viewMode=story) vs [develop (bottom)](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-banner--default&viewMode=story)
image

Banner on `develop` stays centered on larger screens and the one in this PR stays to the left. Mobile seems to look good. Can you look into this issue and then check the compiled CSS file size again?

**Comment by mahoneycm** [comment]
@mejiaj Looks like `grid-container` was setting a max width and auto margins to center it. I added some utility mixins to fix it!

**Review by mejiaj — APPROVED** [review]
Banner (mobile/desktop) looks good! I also compiled using only the banner package and can confirm, no utility layout classes.

**Review by amyleadem — COMMENTED** [review]
@mahoneycm
I tried compiling only the banner package by calling only `usa-banner` and `usa-media-block` in uswds/index.scss. When I do this, I see that the content gets stacked rather than sitting side by side. Could this code be missing the rules from `@include grid-row`? 

image

**Review by amyleadem — CHANGES_REQUESTED** [review]
@mahoneycm 
I turned off all packages except for `usa-banner` and noticed some visual discrepancies between this and the current banner display (only tested in Chrome so far) 

image

image

Let me know if you have questions.

**Comment by mahoneycm** [comment]
Resolved above comment!

File: packages/usa-banner/src/styles/_usa-banner.scss [hunk 1]
```diff
@@ -81,6 +82,25 @@ $banner-icon-close: (
...
+      @include at-media("tablet") {
+        width: 50%
+      }
+    }
+
+    @include at-media("tablet") {
+      @include u-flex-direction("row")
+    }
+  }
 }
```
[the remark this answers is not in the document]
Comment (mahoneycm) [reply packages/usa-banner/src/styles/_usa-banner.scss:104]: Hey @amyleadem, This works great without

**Comment by mahoneycm** [comment]
Thanks to @amyleadem's recommendation above, I realized I overworked my solution and we are able to continue to use the `grid` mixins. Cleaned up my solution to include only what was necessary.

**Review by amyleadem — APPROVED** [review]
Looks good to me! Here is how I tested:
- Turned off all other component Sass except for `usa-media-block` and `usa-banner`
- Checked the appearance in Firefox, Safari, Chrome, and Microsoft for Edge

Visuals matched what I saw on `develop` and the code looks nice and clean. 

Note: I noticed that there are some grid classes inside `usa-banner__header` that aren't addressed in this PR, but the visual presentation still matched `develop` when the layout-grid package was turned off. It seems to be fine, but wanted to flag it in case it has a side effect I am not seeing.

File: packages/usa-banner/src/styles/_usa-banner.scss [hunk 2]
```diff
@@ -81,6 +81,20 @@ $banner-icon-close: (
       margin: 0;
     }
   }
+
+  &.usa-accordion__content > .grid-row {
```
Comment (mejiaj) [inline packages/usa-banner/src/styles/_usa-banner.scss:85]: Do we need to be this specific or can we just target `grid-row` inside of `.usa-banner`?

```diff
-  &.usa-accordion__content > .grid-row {
# Move styles up
+ .usa-banner {
+   .grid-row {
```

**Review by mejiaj — COMMENTED** [review]
Looks good, I was able to test all variants of banner on both mobile and desktop sizes. Just a comment on the specificity of the grid selector.

**Compile test** 
To test compile I did the following:
1. Went to `packages/uswds/_index.scss` and commented everything **except** banner
2. ⚠️ Change `@forward "usa-banner/src/styles"; → @forward "usa-banner";`. _I wonder if the `/src/styles` is intentional or if we should only point to the package._
3. Compiled and verified only `usa-banner` styles
4. Ran `npm start` and did a visual check of all banners

File: packages/usa-banner/src/styles/_usa-banner.scss (hunk 2 above)
Comment (mahoneycm) [reply packages/usa-banner/src/styles/_usa-banner.scss:85]: Looks like there's only the one `.grid-row` class within `usa-banner` so this works! I'll push that up now

**Comment by mahoneycm** [comment]
@thisisdano Ready for final review!

**Review by thisisdano — APPROVED** [review]
Perfect. Really happy to see this change!

**Pull request merged**